va/va.go

```diff
@@ -537,3 +542,26 @@ func (va VAImpl) fetchHTTP(identifier string, token string) ([]byte, string, *ac
 
 	return body, url.String(), nil
 }
+func alpnaddr(addr string) (arpa string, err error) {
+	const hexDigit = "0123456789abcdef"
+	ip := net.ParseIP(addr)
+	if ip == nil {
+		return "", &net.DNSError{Err: "unrecognized address", Name: addr}
+	}
+	if ip.To4() != nil {
+		return string(uint(ip[15])) + "." + string(uint(ip[14])) + "." + string(uint(ip[13])) + "." + string(uint(ip[12])) + ".in-addr.arpa.", nil
```

Comment:
Should there really be the trailing dot (for `in-addr.arpa.`)? I mean, for regular DNS names, the trailing dot isn't there as well, right?

Reply:
techtically the draft itself has trailing dot
For example if the IP address being validated is
   2001:db8::1 the SNI HostName should contain "1.0.0.0.0.0.0.0.0.0.0.0.
   0.0.0.0.0.0.0.0.0.0.0.0.8.b.d.0.1.0.0.2.ip6.arpa.

Reply:
Technically, you should also write `example.com.` instead of `example.com`. But for certificates, one leaves the final dot away.

Reply:
but draft itself explicitly included the dot and have word 'should' so ...

Reply:
Hmm, interesting. The draft explicitly mentions the trailing dot in its example: https://tools.ietf.org/html/draft-ietf-acme-ip-05#section-6
@cpu what do you think, should the dot be there or not?

Reply:
I guess it doesn't really matters, since this isn't put into the certificate, but "only" used in the HTTP protocol while validating the challenge.